mqtt_as/mqtt_as.py

```diff
@@ -406,14 +429,20 @@ def close(self):
             if sz != b"\x02":
                 raise OSError(-1)
             rcv_pid = await self._as_read(2)
+            self.dprint("received PUBACK", rcv_pid)
             self.rcv_pids.add(rcv_pid[0] << 8 | rcv_pid[1])
+            self.dprint("rcv_pids:", self.rcv_pids)
             # Discard old pid's: can arise if a publish fails before PUBACK arrives
-            self.rcv_pids = {x for x in self.rcv_pids if  (x - self.pid) % 65536 < 50}
```

Comment:
Your pruning doesn't work. It removes all pids received and the rcv_pids set was always empty.